### DISCUSSION TITLE
Dynamic page slugs return incorrect parameters in production on Vercel

### DISCUSSION THREAD

**Discussion by rnnyrk** [vercel/next.js#66520] [App Router]
After deploying my project to Vercel, I'm encountering an issue where dynamic page slugs, following the pattern `app/[site]/[locale]/layout.tsx`, are not behaving as expected. Instead of returning the rewritten parameters, they return `[site]` and `[locale]`. This problem only manifests in the production environment on Vercel; it does not occur in either local development or local production builds, which function correctly, therefore it's really hard to reproduce or make an Codesandbox op something similar. 

We have middleware rewriting the incoming url the the param structure. So `/tickets` will be rewritten to `/sitename/nl/tickets`, as you can see in the attached image this is happening correctly (as said locally everything works fine in dev/prod, only on Vercel it breaks). The middl4ware code rewriting the url:

```js
const newPath = `/${host}${
  
url.pathname.includes(locale) ? url.pathname : `/${locale}${url.pathname}`
}`;

url.pathname = newPath;
url.href = `${url.origin}${newPath}`;

const response = NextResponse.rewrite(url);
```

<img width="826" alt="068PZ00000AewqmYAB" src="https://github.com/vercel/next.js/assets/4519142/5b002c40-0833-485b-a35e-fda4a1f906f0">
<img width="1178" alt="068PZ00000AeyEKYAZ" src="https://github.com/vercel/next.js/assets/4519142/c1ffb1e4-e388-45d4-b78a-535a1d23bb34">

So an incoming request for /tickets  is rewritten to /[site]/[locale]/tickets using middleware. This is because we are using a monorepository serving multiple sites from one codebase. 

The rewriting of the url with the correct params is happening correctly on localhost in both development and production builds, but is breaking after deploying to Vercel. After deployment app/[site]/[locale]/layout.tsx doesn't receive the params: {site, locale} as e.g. site = 'sitename' and locale = 'nl' but rather site = [site] and locale = [locale]. So the params are not being rewritten. 

This is only happening on Vercel for the files/routes resulting in a 500:
- app/[site]/[locale]/layout.tsx
- app/[site]/[locale]/tickets/page.tsx
- app/[site]/[locale]/tickets/layout.tsx

On the following routes we have no issue on Vercel and the params are being rewritten as expected:
- app/[site]/[locale]/[...slug]/page.tsx
- app/[site]/[locale]/tickets/[...slug]/page.tsx

We also use pages and app directory alongside each other because we are slowly migrating the project over to app directory. We spend multiple days debugging and trying to fix the issue without any success. Hopefully anyone can point us towards why it doesn't work on specifically Vercel deployment or how we might be able to reproduce it local.

Thans in advance. If there is any more information needed we are happy to supply any.

**Comment by gghdev** [comment]
I'm having the same issue in Next 14.2.5, and _only_ on Vercel, on Node 20. This works correctly in dev and local builds.

In an intercepting route, I'm defining a dynamic parameter like `(...)[companySlug]` and the value I'm getting is literally that, URI encoded: `'%5BcompanySlug%5D'`. There's other dynamic parameter directories beneath and they return the correct values.

Side note: I'm only using `(...)` because `(..)(..)` is also broken.
